python/lsst/scarlet/lite/component.py

```
@@ -395,6 +456,231 @@ def __str__(self):
+        """
+        return FactorizedComponent(
+            bands=self.bands,
+            spectrum=self.spectrum,
+            morph=self.morph,
+            bbox=self.bbox,
+            peak=self.peak,
+            bg_rms=self.bg_rms,
+            bg_thresh=self.bg_thresh,
+            floor=self.floor,
+            monotonicity=self.monotonicity,
+            padding=self.padding,
+            is_symmetric=self.is_symmetric,
+        )
+
+
+class CubeComponent(Component):
+    """Dummy component for a component cube.
+
+    This is duck-typed to a `lsst.scarlet.lite.Component` in order to
+    generate a model from the component but it is currently not functional
+    in that it cannot be optimized, only persisted and loaded.
+
+    If scarlet lite ever implements a component as a data cube,
+    this class can be removed.
+    """
+
+    def __init__(self, model: Image, peak: tuple[int, int]):
+        """Initialization
+
+        Parameters
+        ----------
+        bands :
+        model :
+            The 3D (bands, y, x) model of the component.
+        peak :
+            The `(y, x)` peak of the component.
+        bbox :
+            The bounding box of the component.
+        """
+        super().__init__(model.bands, model.bbox)
+        self._model = model
+        self.peak = peak
+
+    def get_model(self) -> Image:
+        """Generate the model for the source
+
+        Returns
+        -------
+        model :
+            The model as a 3D `(band, y, x)` array.
+        """
+        return self._model
+
+    def resize(self, model_box: Box) -> bool:
+        """Test whether or not the component needs to be resized"""
+        return False
+
+    def update(self, it: int, input_grad: np.ndarray) -> None:
+        """Implementation of unused abstract method"""
+
+    def parameterize(self, parameterization: Callable) -> None:
+        """Implementation of unused abstract method"""
```

Shouldn't this at least warn, if not `raise NotImplementedError`, rather than silently do nothing?